### DISCUSSION TITLE
Question: how to fetch information about a user

### DISCUSSION THREAD

**Discussion by pietervdvn** [neet/masto.js#821] [Q&A] [answered]
Hi,

For my bot, I republish changes that have been made on OpenStreetMap automatically and I do ping people if they have their Mastodon set on OpenStreetMap.

However, I don't want to ping people who indicate that they don't like bots. They do this by setting the "no-bot"-field on their profile. However, I'm struggling to fetch the profile of a user.

I tried `await instance.v1.accounts.fetch("@pietervdvn@en.osm.town")` and `await instance.v1.accounts.fetch("@pietervdvn")`.

What am I doing wrong?

**Comment by neet** [comment, marked as answer]
According to the Mastodon doc:

> `:id`
> REQUIRED String. The ID of the Account in the database.
> https://docs.joinmastodon.org/methods/accounts/#get

Therefore the `id` parameter is the ID of the account which is different from a username. It consists of 18 digits of integer and should look like `109385598812984542`. The value is different in each instance because each instance has its own database.

If you don't know the ID of the target account or don't want to hard-code the ID, you might be able to use [`v1.accounts.lookup`](https://docs.joinmastodon.org/methods/accounts/#lookup) for getting a profile from the username. c.f.

```ts
const acct = await masto.v1.accounts.lookup({
  acct: '@mastodon@mastodon.social',
});
console.log(`ID: ${acct.id}`);
```

I hope this helps

**Reply by pietervdvn** [reply]
Thanks! I initially used `masto.v1.accounts.lookup('@mastodon@mastodon.social');` instead of wrapping it in an object with `{acct: <username>}
